File: sum_tables.js
// sum_tables.js
const { chromium } = require('playwright');

(async () => {
  // List of the pages to visit — seeds 32 through 41 inclusive.
  const urls = [
    'https://sanand0.github.io/tdsdata/js_table/?seed=32',
    'https://sanand0.github.io/tdsdata/js_table/?seed=33',
    'https://sanand0.github.io/tdsdata/js_table/?seed=34',
    'https://sanand0.github.io/tdsdata/js_table/?seed=35',
    'https://sanand0.github.io/tdsdata/js_table/?seed=36',
    'https://sanand0.github.io/tdsdata/js_table/?seed=37',
    'https://sanand0.github.io/tdsdata/js_table/?seed=38',
    'https://sanand0.github.io/tdsdata/js_table/?seed=39',
    'https://sanand0.github.io/tdsdata/js_table/?seed=40',
    'https://sanand0.github.io/tdsdata/js_table/?seed=41'
  ];

  const browser = await chromium.launch({ headless: true });
  const context = await browser.newContext();
  const page = await context.newPage();

  let grandTotal = 0;

  for (const url of urls) {
    console.log(`Visiting: ${url}`);
    try {
      await page.goto(url, { waitUntil: 'networkidle', timeout: 30000 });
    } catch (err) {
      console.error(`Failed to load ${url}: ${err.message}`);
      continue;
    }

    // Evaluate in page: gather all table cell text and extract numbers robustly.
    const pageSum = await page.evaluate(() => {
      // helper to extract numbers from a string, returns array of numeric strings
      function extractNumbers(str) {
        if (!str || typeof str !== 'string') return [];
        // Match numbers like: 1,234.56 or -1,234 or 1234 or .56 or -0.5
        const matches = str.match(/-?\d[\d,]*\.?\d*|-?\.\d+/g);
        return matches || [];
      }

      // Collect text from table cells (td and th to be safe)
      const cells = Array.from(document.querySelectorAll('table td, table th'));
      let sum = 0;
      for (const cell of cells) {
        const text = cell.innerText || cell.textContent || '';
        const nums = extractNumbers(text);
        for (const n of nums) {
          // remove comma thousands separators then parse
          const cleaned = n.replace(/,/g, '');
          const parsed = parseFloat(cleaned);
          if (!isNaN(parsed)) sum += parsed;
        }
      }
      return sum;
    });

    console.log(`Page sum for ${url}: ${pageSum}`);
    grandTotal += pageSum;
  }

  await browser.close();

  // Print the grand total with a consistent number of decimals (if needed)
  // If you expect integers, you can print as integer. We'll show up to 6 decimal places if necessary.
  const formatted = Number.isInteger(grandTotal) ? grandTotal.toString() : grandTotal.toFixed(6);
  console.log('========================');
  console.log('GRAND TOTAL (sum of all numbers in all tables across pages):');
  console.log(formatted);
  console.log('========================');

  // Exit with success code
  process.exit(0);
})();

File: sum_tables.py
from playwright.sync_api import sync_playwright

def sum_js_tables(base_url: str, seeds: list[str]) -> int:
    """
    Uses Playwright to visit multiple pages, finds all numbers in tables,
    and returns their sum.

    Args:
        base_url (str): The base URL for the pages.
        seeds (list[str]): A list of seeds to generate the full URLs.

    Returns:
        int: The total sum of all numbers found in the tables.
    """
    total_sum = 0

    with sync_playwright() as p:
        # Launch a headless browser (headless=True means no UI is shown)
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()

        for seed in seeds:
            url = f"{base_url}?seed={seed}"
            print(f"Navigating to {url}...")

            try:
                page.goto(url, wait_until="domcontentloaded")

                # Use a locator to find all table cells (td)
                # Playwright's locators auto-wait, which is robust
                cells = page.locator("td")

                # Get the count for logging
                cell_count = cells.count()
                print(f"  Found {cell_count} cells.")

                # Iterate through each cell and add its value to the sum
                for i in range(cell_count):
                    cell_text = cells.nth(i).inner_text()
                    if cell_text.isdigit():
                        total_sum += int(cell_text)

            except Exception as e:
                print(f"  An error occurred on page {url}: {e}")
                continue # Move to the next seed

        # Close the browser instance
        browser.close()

    return total_sum


if __name__ == "__main__":
    # --- Replace these with the values from your question ---
    # The problem specifies a range of seeds to use.
    # For example, if the start seed is 76 and the end seed is 85:
    start_seed = 76
    end_seed = 85
    SEEDS_TO_PROCESS = [str(i) for i in range(start_seed, end_seed + 1)]
    BASE_URL = "https://sanand0.github.io/tdsdata/js_table/"
    # ---------------------------------------------------------

    print(f"Processing seeds: {SEEDS_TO_PROCESS}")
    final_sum = sum_js_tables(BASE_URL, SEEDS_TO_PROCESS)

    print("\n--- Result ---")
    print(f"The total sum of all numbers in the tables is: {final_sum}")
    print("----------------\n")
